Luke7787/ImmaculateInventors
# Feature/create account

Summary:
- Created UI for Create Account Page
- Added validation for the form 
- Created backend GET request to check if username is unique for means of validation
- Efficiently cleaned up all dependency errors with the linting process

Link to any specs used:
[Spec](https://www.figma.com/file/AE3t3fUj0nMRRtUQX3BBbT/UI-Design---Sign-In-Modal?type=design&node-id=0%3A1&mode=design&t=vSYGX9U3DuFaNLq7-1)
How to test:
- In root, `npm start`
- In express-backend, `npm run dev`
- Click sign in button on top right
- Click "Create an account here"
- Hit "submit"
   - Verify that all fields receive an error
- Fill in first name, last name
- For email:
   - Type an invalid email (e.g. "hello")
   - Verify that "Email address is not valid" error message appears
- Fill in country, state, city, zip
- For username:
   - Use existing username, so "Adam" and click out of the box
   - verify that an error message appears saying that the username is already taken
- For password: verify that each of these receive an error message, besides the last one
   - Insert a password under 10 chars   
   - Insert a password 10+ chars, no special chars
   - Insert a password 10+ chars, special chars, no uppercase char
   - Insert a password 10+ chars, special char, uppercase char, no number
   - Insert a password 10+ chars, special char, uppercase char, number
- For confirm password:
   - Insert a non-matching password
      - Verify that you get an error
   - Insert a matching password
      - Verify no error is thrown

- Click submit
   - Verify that all data is cleared and the console displays a log of your data

Notes:
N/A
